herald/base.py

```diff
@@ -57,6 +56,20 @@ def get_verbose_name(cls):
     def get_class_path(cls):
         return "{}.{}".format(cls.__module__, cls.__name__)
 
+    def preview(self, render_type=None):
+        """
+        Renders a notification for user to preview a notification before sending it out.
+        This method only renders and does not send the email.
+        """
+
+        # Check if render_type is valid
+        if render_type is None or render_type not in self.render_types:
+            raise ValueError(
+                f"{render_type} is not a valid render type. Must be among {self.render_types}"
```

Comment:
@HieuPham9720 I would suggest using string interpolation here for external error tools like Sentry.

```python
ValueError("%s is not a valid render type. Must be among %s" % (render_type, self.render_types))
```